runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/ConfigurationManager.java

```diff
@@ -155,7 +160,13 @@ public NamespaceConfig parse(
             {
                 binding.id = namespace.resolveId.applyAsLong(binding.name);
                 binding.entryId = namespace.resolveId.applyAsLong(binding.entry);
+                binding.typeId = namespace.resolveId.applyAsLong(binding.type);
+                binding.kindId = namespace.resolveId.applyAsLong(binding.kind.name().toLowerCase());
+                Binding b = bindingByType.apply(binding.type);
+                binding.originTypeId = namespace.resolveId.applyAsLong(b.originType(binding.kind));
+                binding.routedTypeId = namespace.resolveId.applyAsLong(b.routedType(binding.kind));
                 binding.resolveId = namespace.resolveId;
+                writeBindingInfo.accept(binding);
```

Comment:
If we're doing this in the engine, perhaps we don't need these additional transient fields that are single use.

Reply:
OK